classes/external/ai/generate_ai_text.php

```diff
@@ -111,12 +121,14 @@ public  static function execute(int $assignment_id, string $text, int $step = 1)
         }
         catch (Exception $exception)
         {
-            $error = new moodle_traceable_exception('error_generate_ai_text_api', $exception);
-            $archive->rollback(
-                $text,
-                $error
-            );
-            throw $error;
+            try
+            {
+                $transaction->rollback($exception);
+            }
+            catch (\Exception $rollback_exception) {}
```

Comment:
:laughing:

Reply:
Yeah Moodle throw an exception  when we call ``rollback``. That is very mess up...